File: frontend/lib/voicePresets.ts
/**
 * Display mirror of backend/app/services/voice_presets.py.
 *
 * Only Aiden and Ryan are native-English Qwen3-TTS presets; the other seven are
 * Chinese, Japanese or Korean. We label rather than hide, and preview each voice
 * in its own language — an English sample from a Chinese speaker is exactly the
 * bad impression the labelling prevents.
 *
 * Scope: this makes the VOICE layer multi-language. Script generation, the studio
 * UI and subtitles remain English-only.
 */

export type PresetLanguage = 'en' | 'zh' | 'ja' | 'ko';

export interface VoicePreset {
    speaker: string;
    label: string;
    language: PresetLanguage;
    languageLabel: string;
    gender: 'male' | 'female';
    accent?: string;
    gloss: string;
}

const GLOSS = 'Welcome to Aria Appeal.';

export const DEFAULT_PRESET = 'Aiden';

export const VOICE_PRESETS: VoicePreset[] = [
    { speaker: 'Aiden', label: 'Aiden — Male', language: 'en', languageLabel: 'English', gender: 'male', gloss: GLOSS },
    { speaker: 'Ryan', label: 'Ryan — Male', language: 'en', languageLabel: 'English', gender: 'male', gloss: GLOSS },
    { speaker: 'Vivian', label: 'Vivian — Female', language: 'zh', languageLabel: 'Chinese', gender: 'female', gloss: GLOSS },
    { speaker: 'Serena', label: 'Serena — Female', language: 'zh', languageLabel: 'Chinese', gender: 'female', gloss: GLOSS },
    { speaker: 'Uncle_Fu', label: 'Uncle Fu — Male', language: 'zh', languageLabel: 'Chinese', gender: 'male', gloss: GLOSS },
    { speaker: 'Dylan', label: 'Dylan — Male, Beijing', language: 'zh', languageLabel: 'Chinese', gender: 'male', accent: 'Beijing', gloss: GLOSS },
    { speaker: 'Eric', label: 'Eric — Male, Sichuan', language: 'zh', languageLabel: 'Chinese', gender: 'male', accent: 'Sichuan', gloss: GLOSS },
    { speaker: 'Ono_Anna', label: 'Ono Anna — Female', language: 'ja', languageLabel: 'Japanese', gender: 'female', gloss: GLOSS },
    { speaker: 'Sohee', label: 'Sohee — Female', language: 'ko', languageLabel: 'Korean', gender: 'female', gloss: GLOSS },
];

// English first — it is the only language the rest of the product speaks.
const LANGUAGE_ORDER: PresetLanguage[] = ['en', 'zh', 'ja', 'ko'];

export interface PresetGroup {
    language: PresetLanguage;
    label: string;
    presets: VoicePreset[];
}

export const PRESET_GROUPS: PresetGroup[] = LANGUAGE_ORDER.map(language => ({
    language,
    label: VOICE_PRESETS.find(p => p.language === language)!.languageLabel,
    presets: VOICE_PRESETS.filter(p => p.language === language),
}));

const SPEAKERS = new Set(VOICE_PRESETS.map(p => p.speaker));

export function isPresetSpeaker(value: string): boolean {
    return SPEAKERS.has(value);
}

export function getPreset(speaker: string): VoicePreset | undefined {
    return VOICE_PRESETS.find(p => p.speaker === speaker);
}

File: backend/app/services/voice_presets.py
"""The nine Qwen3-TTS CustomVoice preset speakers, with the language each is
native to.

Per the Qwen3-TTS-12Hz-1.7B-CustomVoice model card only Aiden and Ryan are
English; the rest are Chinese, Japanese or Korean and read English badly. Qwen's
own guidance is to use each speaker's native language, so previews greet in the
speaker's language and carry an English gloss for the caption.

This is the source of truth for synthesis. frontend/lib/voicePresets.ts mirrors
the display fields; test_voice_presets.py guards both against drift with the
engine's PRESET_SPEAKERS list.
"""
from dataclasses import dataclass
from typing import Optional

ENGLISH_DEFAULT = "Aiden"

_GREETINGS = {
    "en": "Welcome to Aria Appeal.",
    "zh": "欢迎使用 Aria Appeal。",
    "ja": "Aria Appeal へようこそ。",
    "ko": "Aria Appeal에 오신 것을 환영합니다.",
}

_ENGLISH_GLOSS = "Welcome to Aria Appeal."


@dataclass(frozen=True)
class VoicePreset:
    speaker: str
    language: str          # ISO-639-1: en / zh / ja / ko
    language_label: str    # human label for the picker group heading
    gender: str            # "female" | "male"
    accent: Optional[str] = None  # regional note, e.g. "Beijing"

    @property
    def greeting(self) -> str:
        """The preview line, in this speaker's own language."""
        return _GREETINGS[self.language]

    @property
    def gloss(self) -> str:
        """English caption shown next to a non-English preview."""
        return _ENGLISH_GLOSS

    @property
    def label(self) -> str:
        gender = self.gender.capitalize()
        base = f"{self.speaker.replace('_', ' ')} — {gender}"
        return f"{base} ({self.language_label}, {self.accent})" if self.accent \
            else f"{base} ({self.language_label})"


PRESETS = (
    VoicePreset("Aiden", "en", "English", "male"),
    VoicePreset("Ryan", "en", "English", "male"),
    VoicePreset("Vivian", "zh", "Chinese", "female"),
    VoicePreset("Serena", "zh", "Chinese", "female"),
    VoicePreset("Uncle_Fu", "zh", "Chinese", "male"),
    VoicePreset("Dylan", "zh", "Chinese", "male", accent="Beijing"),
    VoicePreset("Eric", "zh", "Chinese", "male", accent="Sichuan"),
    VoicePreset("Ono_Anna", "ja", "Japanese", "female"),
    VoicePreset("Sohee", "ko", "Korean", "female"),
)

_BY_SPEAKER = {p.speaker: p for p in PRESETS}


def get_preset(speaker: str) -> Optional[VoicePreset]:
    return _BY_SPEAKER.get(speaker)
